scripts/beachball/customRenderers.ts

```diff
@@ -33,18 +36,25 @@ export async function renderHeader(renderInfo: PackageChangelogRenderInfo): Prom
 
 export async function renderEntry(entry: ChangelogEntry): Promise<string> {
   // Link to the PR for this changelog entry (or the commit if PR isn't found)
-  const prNumber = await _getPrNumber(entry);
+  const prNumber = await _getPrNumber(entry.commit);
   const commitLink = prNumber
     ? `[PR #${prNumber}](${repoUrl}/pull/${prNumber})`
     : `[commit](${repoUrl}/commit/${entry.commit})`;
   return `- ${entry.comment} (${commitLink} by ${entry.author})`;
 }
 
-async function _getPrNumber(entry: ChangelogEntry): Promise<number | undefined> {
+async function _getPrNumber(commit: string): Promise<number | undefined> {
+  if (!prNumbers.has(commit)) {
+    prNumbers.set(commit, await _lookUpPrNumber(commit));
+  }
+  return prNumbers.get(commit);
+}
+
+async function _lookUpPrNumber(commit: string): Promise<number | undefined> {
   // Look for (presumably) the PR number at the end of the first line of the commit
   try {
     // Get the actual commit message which should contain the PR number
-    const logResult = spawnSync('git', ['log', '--pretty=format:%s', '-n', '1', entry.commit]);
+    const logResult = spawnSync('git', ['log', '--pretty=format:%s', '-n', '1', commit]);
```

Comment:
I'm probably missing something obvious, but which line or lines in the `try` block will ever throw? I don't see how we could end up in the `catch` block.

Reply:
Possibly `git log` but I'm not exactly sure how that would happen (IIRC it would only be if there was an error starting the command, not if the command exited non-0).

I should probably just close this PR because we're likely going to be re-thinking our approach to releases and release notes.